src/julia.h

```diff
@@ -627,6 +633,27 @@ typedef struct _jl_gcframe_t {
 
 #define jl_pgcstack (jl_get_ptls_states()->pgcstack)
 
+#ifdef __clang_analyzer__
+
+// When running with the analyzer make these real function calls, that are
+// easier to detect in the analyzer
+extern void JL_GC_PUSH1(void *) JL_NOTSAFEPOINT;
+extern void JL_GC_PUSH2(void *, void *) JL_NOTSAFEPOINT;
+extern void JL_GC_PUSH3(void *, void *, void *)  JL_NOTSAFEPOINT;
+extern void JL_GC_PUSH4(void *, void *, void *, void *)  JL_NOTSAFEPOINT;
+extern void JL_GC_PUSH5(void *, void *, void *, void *, void *)  JL_NOTSAFEPOINT;
+extern void _JL_GC_PUSHARGS(jl_value_t **, size_t) JL_NOTSAFEPOINT;
+// This is necessary, because otherwise the analyzer considers this undefined
+// behavior and terminates the exploration
+#define JL_GC_PUSHARGS(rts_var, n)     \
+  rts_var = (jl_value_t **)alloca(sizeof(void*) * n); \
+  memset(rts_var,0,sizeof(void*) * n); \
+  _JL_GC_PUSHARGS(rts_var, n);
```

Comment:
Just curious, what the UB here? Uninitialized variable?

Reply:
Yes, read of uninitialized memory (either rts_var itself, or the contained pointers if not memset to 0).

Reply:
I've personally been confused by it a few times. I would support making it `JL_GC_PUSHARGS(&var, n)` instead.

Reply:
Fine with me, but independent of this change ;)

Reply:
Sure